pkg/controller/batchrelease/control/canarystyle/deployment/canary.go

```diff
@@ -179,9 +191,22 @@ func filterCanaryDeployment(ds []*apps.Deployment, template *corev1.PodTemplateS
 	if template == nil {
 		return ds[0]
 	}
+	dTemplate := template.DeepCopy()
 	for _, d := range ds {
-		// todo, remove the canary pod metadata
-		if util.EqualIgnoreHash(template, &d.Spec.Template) {
+		// remove the canary pod metadata
+		dClone := d.DeepCopy()
+		if release.Spec.ReleasePlan.PatchPodTemplateMetadata != nil {
+			patch := release.Spec.ReleasePlan.PatchPodTemplateMetadata
+			for k := range patch.Labels {
+				delete(dClone.Spec.Template.Labels, k)
+				delete(dTemplate.Labels, k)
+			}
+			for k := range patch.Annotations {
+				delete(dClone.Spec.Template.Annotations, k)
+				delete(dTemplate.Annotations, k)
+			}
+		}
+		if util.EqualIgnoreHash(dTemplate, &dClone.Spec.Template) {
```

Comment:
dClone is deepcopy twice here, can we make a modified EqualIgnoreHash func e.g. EqualIgnoreMetadata(template1, template2 *v1.PodTemplateSpec, labels []string, annotations []string)